### PR TITLE
Fixed precision tensor

### PR DESCRIPTION
**Pull request by rasswanth-s**
## Description
Implementation of FixedPrecisionTensor.

## Affected Dependencies
SMPC and DP code

## How has this been tested?
Tests to be added.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: packages/syft/src/syft/core/tensor/lazy_repeat_array.py [hunk 1]
```diff
@@ -158,6 +190,15 @@ def __le__(self, other: Any) -> lazyrepeatarray:  # type: ignore
 
         return self <= other
 
+    def concatenate(self, other: lazyrepeatarray, *args, **kwargs) -> lazyrepeatarray:
+        if not isinstance(other, lazyrepeatarray):
+            raise NotImplementedError
+
+        dummy_res = dummy_res = np.concatenate(
```
Comment (gmuraru) [inline packages/syft/src/syft/core/tensor/lazy_repeat_array.py:197]: MAYBE: Remove one `dummy_res`

File: packages/syft/src/syft/core/tensor/lazy_repeat_array.py [hunk 2]
```diff
@@ -62,8 +65,12 @@ def __add__(self, other: Any) -> lazyrepeatarray:
 
         if self.data.shape == other.data.shape:
             return self.__class__(data=self.data + other.data, shape=self.shape)
+        else:
+            # Rasswanth : Fix after addition of integration tests.
+            print("Lazy Repeat adding with mismatched shapes")
```
Comment (gmuraru) [inline packages/syft/src/syft/core/tensor/lazy_repeat_array.py:70]: Q: Can't we use the `logger` here? (maybe add the `info`)

File: packages/syft/src/syft/core/tensor/smpc/context.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# future
+from __future__ import annotations
+
+# stdlib
+from typing import Optional
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
```
[the remark this answers is not in the document]
Comment (rasswanth-s) [reply packages/syft/src/syft/core/tensor/smpc/context.py:8]: Yep

File: packages/syft/src/syft/core/tensor/lazy_repeat_array.py (hunk 2 above)
Comment (rasswanth-s) [reply packages/syft/src/syft/core/tensor/lazy_repeat_array.py:70]: This I will plan to remove before merge,I did this quick fix for a demo.

File: packages/syft/src/syft/core/tensor/lazy_repeat_array.py (hunk 1 above)
Comment (rasswanth-s) [reply packages/syft/src/syft/core/tensor/lazy_repeat_array.py]: removed.

**Review by madhavajay — APPROVED** [review]
Awesome work this is one of those historical event PRs. 🚀🌕

**Pull request merged**